include/string_utils.h

```diff
@@ -311,18 +311,31 @@ enum class UnicodeFallback {
 	Box
 };
 
+// Specifies how to interpret characters 0x00-0x1f and 0x7f in DOS strings
+enum class DosStringType {
+	// String contains control codes (new line, tabulation, delete, etc.)
+	WithControlCodes,
+	// String does not have any codes characters, consider all characters
+	// as screen codes
+	ScreenCodesOnly,
+	// String should not contain characters mentioned above
+	NoSpecialCharacters
+};
+
 // Convert the UTF-8 string to the format intended for display inside emulated
 // environment, or vice-versa. Code page '0' means a pure 7-bit ASCII. Functions
 // without 'code_page' parameters uses current DOS code page.
 // Return value 'false' means there were problems with string decoding or
 // rendering, but the overall output should still be sane.
 bool utf8_to_dos(const std::string& in_str, std::string& out_str,
-                 const UnicodeFallback fallback);
+                 const DosStringType string_type, const UnicodeFallback fallback);
 bool utf8_to_dos(const std::string& in_str, std::string& out_str,
+                 const DosStringType string_type,
                  const UnicodeFallback fallback, const uint16_t code_page);
-void dos_to_utf8(const std::string& in_str, std::string& out_str);
 void dos_to_utf8(const std::string& in_str, std::string& out_str,
-                 const uint16_t code_page);
+                 const DosStringType string_type);
+void dos_to_utf8(const std::string& in_str, std::string& out_str,
```

Comment:
Instead of the old school "out var" style, you could never mutate passed in string args, but return `std::optional<std::string>` instead. This would improve the API.

Reply:
False does not mean the conversion totally failed, just that there were some problems - but the string should be usable. `std::optional` is not suitable here.

Reply:
@FeralChild64 Ok, that's a bit unexpected. Is this behaviour documented in the header files,?

Reply:
@johnnovak Yes, it's documented.

BTW, after some more checking it seems that API rework is easier than I thought. I'm preparing a follow-up PR, with some Unicode refactoring, https://github.com/dosbox-staging/dosbox-staging/pull/3797

Reply:
Great, thanks @FeralChild64 👍🏻